hail/src/main/scala/is/hail/expr/ir/functions/IntervalFunctions.scala

```diff
@@ -125,6 +125,17 @@ object IntervalFunctions extends RegistryFunctions {
             interval1.isAboveOnNonempty(interval2))
         )
     }
+
+    registerIR("sortedNonOverlappingIntervalsContain",
+      TArray(TInterval(tv("T"))), tv("T"), TBoolean()) { case (intervals, value) =>
+      val uid = genUID()
+      val uid2 = genUID()
+      Let(uid, LowerBoundOnOrderedCollection(intervals, value, onKey = true),
```

Comment:
Does this typecheck? `LowerBoundOnOrderedCollection` with `onKey = true` requires the element type to be `TStruct`.

Reply:
oh, I do need to change TypeCheck.

I guess it was never called since this was in an ApplyIR?